### PR TITLE
feat: remove var ident calc in codegen process

### PR DESCRIPTION
**Pull request by zhusjfaker**

### CODE REVIEW

File: crates/style/tests/css/select.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+use rspack_style::css::select::NewSelector;
+use rspack_style::extend::string::StringExtend;
+
+#[test]
+fn test_css_select_var_error_parse() {
+  let ss = "@{abc} .b";
+  let mut obj = NewSelector::new(ss.to_string().to_char_vec(), None, None, None, None);
+  let mut has_error = 0;
+  match obj.parse() {
```
Comment (IWANABETHATGUY) [inline crates/style/tests/css/select.rs:9]: `assert!(obj.parse().is_err())` would be enough

File: crates/style/tests/css/arco.rs [hunk 2]
```diff
@@ -7,3 +7,20 @@ fn arco_global_css_render() {
...
+    let file = entry.unwrap().path();
+    if file.is_file() && file.extension().unwrap().to_str().unwrap() == "css" {
+      let css_file = file.to_str().unwrap();
+      let (css, js) = app.render(css_file).unwrap();
+      println!("path -> \n {}", css_file);
+      println!("css -> \n {}", css);
+      println!("js -> \n {}", js);
+    }
+  }
+  assert_eq!(1, 1);
```
Comment (IWANABETHATGUY) [inline crates/style/tests/css/arco.rs:25]: What is the meaning of this assertion here?

Comment (zhusjfaker) [reply crates/style/tests/css/arco.rs]:
```

#[test]
fn arco_all_css_render() {
  let mut res = vec![];
  let dir = path_resolve("assets/arco-pro-css");
  let app = Application::default();
  for entry in std::path::Path::new(&dir).read_dir().unwrap() {
    let file = entry.unwrap().path();
    if file.is_file() && file.extension().unwrap().to_str().unwrap() == "css" {
      let css_file = file.to_str().unwrap();
      let css_render_res = app.render(css_file);
      println!("path -> \n {}", css_file);
      res.push(css_render_res.is_ok());
    }
  }
  assert!(!res.contains(&false))
}
```

File: crates/style/tests/css/select.rs (hunk 1 above)
Comment (zhusjfaker) [reply crates/style/tests/css/select.rs]:
```
#[test]
fn test_css_select_var_error_parse() {
  let ss = "@{abc} .b";
  let mut obj = NewSelector::new(ss.to_string().to_char_vec(), None, None, None, None);
  assert!(obj.parse().is_err());
}
```

File: crates/style/tests/css/arco.rs (hunk 2 above)
Comment (IWANABETHATGUY) [reply crates/style/tests/css/arco.rs]: You don't need to paste the whole changed test, which may waste your time. I will look at it myself after you address these issues, anyway thanks for your reply.

**Pull request merged**